app/models/protocol.rb

```diff
@@ -306,27 +301,20 @@ def primary_pi_exists
     errors.add(:base, "Only one Primary PI is allowed. Please ensure that only one exists") if project_roles.select { |pr| pr.role == 'primary-pi'}.count > 1
   end
 
-  def role_for identity
-    project_roles.detect{|pr| pr.identity_id == identity.id}.try(:role)
+  def role_for(identity)
+    project_roles.find_by(identity_id: identity.id).try(:role)
   end
 
-  def role_other_for identity
-    project_roles.detect{|pr| pr.identity_id == identity.id}.try(:role)
+  def role_other_for(identity)
+    role_for(identity)
```

Comment:
It is, but I'm preserving the Protocol model's interface in this pull request. Just wanting to make memory usage, speed and style improvements. Once this PR is vetted however, we should totally rip out redundant methods and the like.